assembl/static2/js/app/components/navbar/navbar.jsx

```diff
@@ -0,0 +1,189 @@
+};
+
+const constantToCamelCase = (string) => {
+  return string.toLowerCase().replace(/_[a-z]/g, (match) => {
+    return match[1].toUpperCase();
+  });
+};
+
+const sectionSlug = (sectionType) => {
+  return constantToCamelCase(sectionType === 'HOMEPAGE' ? 'HOME' : sectionType);
+};
+
+const sectionURL = ({ sectionType, url }, slug, phaseIdentifier) => {
+  return url || `${get(sectionSlug(sectionType), { slug: slug, phase: phaseIdentifier })}`;
+};
+
+const SectionLink = ({ section, options: { slug, currentPhaseIdentifier } }) => {
+  const { title } = section;
+  return (
+    <Link
+      to={sectionURL(section, slug, currentPhaseIdentifier)}
+      className="navbar-menu-item pointer"
+      activeClassName="active"
+      data-text={title}
+    >
+      {title}
+    </Link>
+  );
+};
+
+const createDisplayModal = ({ debate, i18n }) => {
+  return () => {
+    const slug = { slug: getDiscussionSlug() };
+    const { timeline } = debate.debateData;
+    const { locale } = i18n;
+    const currentPhaseIdentifier = getCurrentPhaseIdentifier(timeline);
+    const phaseName = getPhaseName(timeline, currentPhaseIdentifier, locale).toLowerCase();
+    const body = <Translate value="redirectToV1" phaseName={phaseName} />;
+    const button = { link: get('oldDebate', slug), label: I18n.t('home.accessButton'), internalLink: false };
+    displayModal(null, body, true, null, button, true);
+    setTimeout(() => {
+      window.location = get('oldDebate', slug);
+    }, 6000);
+  };
+};
+
+const mapDebateSectionToElement = (debateSection, options) => {
+  const { title } = debateSection;
+  const { phaseContext, displayDebateModal } = options;
+  const key = sectionKey(debateSection);
+  switch (phaseContext) {
+  case 'modal':
+    return (
+      <div key={key} onClick={displayDebateModal} className="navbar-menu-item pointer" data-text={title}>
+        {title}
+      </div>
+    );
+  case 'old':
+    return (
+      <a key={key} className="navbar-menu-item pointer" href={get('oldDebate', { slug: options.slug })} data-text={title}>
+        {title}{' '}
+      </a>
+    );
+  default:
+    return <SectionLink key={key} section={debateSection} options={options} />;
+  }
+};
+
+const mapSectionToElement = (section, options) => {
+  switch (section.sectionType) {
+  case 'DEBATE':
+    return mapDebateSectionToElement(section, options);
+  default:
+    return <SectionLink key={sectionKey(section)} section={section} options={options} />;
+  }
+};
+
+const sectionMapper = (options) => {
+  return (section) => {
+    return mapSectionToElement(section, options);
+  };
+};
+
+const phaseContext = (timeline, phase) => {
+  const isSeveralPhases = isSeveralIdentifiers(timeline);
+  if (phase.isRedirectionToV1) {
+    if (isSeveralPhases) {
+      return 'modal';
+    }
+    return 'old';
+  }
+  return 'new';
+};
+
+class AssemblNavbar extends React.PureComponent {
+  setFlatWidth = (newWidth) => {
+    this.setState({ flatWidth: newWidth });
+  };
```

Comment:
I like empty lines between methods in a class, it improves readability. I don't understand it is not part of our eslint rules (maybe there is no eslint rule for this). Any opinions @vincentfretin @paolina92 @ayyazdaniaryan @n0izn0iz  ?

Reply:
yes, I want newline between methods. It looks like a rule was implemented the 2 october https://github.com/eslint/eslint/commit/ee99876ccc76585b0af2c8673d7eb25bb0b972fa should be in eslint 4.12.1

Reply:
that was released today, lol

Reply:
`"lines-between-class-members": 2` show 150 errors :) Hopefully it's fixable automatically with `--fix` We can add the rule and fix all files at the same time as #242

Reply:
I always but lines between my methods in javascript. It's a verbose enough language, don't like to clutter methods together.

Reply:
So, should I edit or wait for autofix ?

Reply:
Don't bother to edit, we can wait for autofix.